### PR TITLE
Use structured initialization in two more places

### PR DESCRIPTION
**Pull request by mrc0mmand**
And squash one unnecessary assignment.

Addresses feedback from https://github.com/systemd/systemd/pull/27803.

### CODE REVIEW

**Review by yuwata — APPROVED** [review]

File: src/journal-remote/journal-remote-write.c [hunk 1]
```diff
@@ -28,19 +28,20 @@ int writer_new(RemoteServer *server, Writer **ret) {
...
 
-        w->n_ref = 1;
-        w->metrics = server->metrics;
+        *w = (Writer) {
+                .n_ref = 1,
+                .metrics = server->metrics,
+                .server = server,
+        };
 
         w->mmap = mmap_cache_new();
```
Comment (dtardon) [inline src/journal-remote/journal-remote-write.c:41]: Please move this to the initializer block too.

File: src/libsystemd-network/fuzz-dhcp-server.c [hunk 2]
```diff
@@ -19,30 +19,32 @@ ssize_t sendmsg(int sockfd, const struct msghdr *msg, int flags) {
...
+                .chaddr = { 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3 },
+                .expiration = UINT64_MAX,
+                .gateway = server_address->s_addr,
+                .hlen = ETH_ALEN,
+                .htype = ARPHRD_ETHER,
+
+                .client_id.length = 2,
+        };
+
+        lease->client_id.data = new(uint8_t, lease->client_id.length);
```
Comment (dtardon) [inline src/libsystemd-network/fuzz-dhcp-server.c:41]: Ditto

File: src/journal-remote/journal-remote-write.c (hunk 1 above)
Comment (mrc0mmand) [reply src/journal-remote/journal-remote-write.c:41]: I prefer to keep the allocation closely together with the error check. Looking at existing code it seems to be split on this one as well, where, for example, logind-brightness/sysctl does the "inline" alloc, whereas logind-{inhibit,seat,session}, netlink-genl.c, etc. keeps the alloc together with the error check.

**Comment by poettering** [comment]
lgtm

**Review by bluca — APPROVED** [review]

**Pull request merged**